### ISSUE TITLE
Pattern Placements Fusion bug

### ISSUE THREAD

**Issue by jaypantone** [PermutaTriangle/Tilings#164]
Traceback:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-2f516ba3e130> in <module>
     24 # pack = pack.add_inferral(SubobstructionInferralStrategy())
     25 css = TileScope(basis, pack)
---> 26 tree = css.auto_search(status_update=30)
     27
     28 # for i in range(10):

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/comb_spec_searcher.py in auto_search(self, **kwargs)
    565                     last_label = label
    566                 if not self.ruledb.is_verified(label):
--> 567                     self._expand(comb_class, label, strategies, inferral)
    568                 if time.time() - expansion_start > max_expansion_time:
    569                     break

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/comb_spec_searcher.py in _expand(self, comb_class, label, strategies, inferral)
    154                     comb_class, strategy_generator, label
    155                 ):
--> 156                     self._add_rule(start_label, end_labels, rule)
    157
    158     def _rules_from_strategy(

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/comb_spec_searcher.py in _add_rule(self, start_label, end_labels, rule)
    292             # this must be a verification strategy!
    293             assert isinstance(rule, VerificationRule), rule.formal_step
--> 294         self.ruledb.add(start_label, tuple(cleaned_end_labels), rule)
    295
    296     def _add_empty_rule(self, label: int) -> None:

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/rule_db.py in add(self, start, ends, rule)
     69             isinstance(rule, Rule)
     70             and len(ends) == 1
---> 71             and rule.constructor.is_equivalence()
     72         ):
     73             self.set_equivalent(start, ends[0])

~/Dropbox/Research/Active/2017-44-ATRAP/repos/comb_spec_searcher/comb_spec_searcher/strategies/rule.py in constructor(self)
    313         if self._constructor is None:
    314             self._constructor = self.strategy.constructor(
--> 315                 self.comb_class, self.children
    316             )
    317             if self._constructor is None:

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/strategies/fusion.py in constructor(self, comb_class, children)
    463                 raise StrategyDoesNotApply("Strategy does not apply")
    464         return FusionConstructor(
--> 465             self._fuse_parameter(comb_class),
    466             self.extra_parameters(comb_class, children)[0],
    467             *self.left_right_both_sided_parameters(comb_class),

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/strategies/fusion.py in _fuse_parameter(self, comb_class)
    514         child = algo.fused_tiling()
    515         fuse_assumption = algo.new_assumption()
--> 516         return child.get_parameter(fuse_assumption)
    517
    518     def formal_step(self) -> str:

~/Dropbox/Research/Active/2017-44-ATRAP/repos/Tilings/tilings/tiling.py in get_parameter(self, assumption)
   1071
   1072     def get_parameter(self, assumption: TrackingAssumption) -> str:
-> 1073         idx = self._assumptions.index(assumption)
   1074         return "k_{}".format(idx)
   1075

ValueError: tuple.index(x): x not in tuple
```

To create (takes about half an hour):
```python
from tilings.tilescope import TileScope, TileScopePack
basis = '0132_0213'
pack = TileScopePack.pattern_placements().make_fusion()
css = TileScope(basis, pack)
spec = css.auto_search(status_update=30)
```

**Comment by jaypantone** [comment]
More info. When the error appears from
```python
    464         return FusionConstructor(
--> 465             self._fuse_parameter(comb_class),
    466             self.extra_parameters(comb_class, children)[0],
    467             *self.left_right_both_sided_parameters(comb_class),
```
the values are:
`comb_class`:
```
+-+-+-+-+-+
| | |2|2| |
+-+-+-+-+-+
|1| | | |/|
+-+-+-+-+-+
| |1| | | |
+-+-+-+-+-+
/: Av(10)
1: Av(021)
2: Av(021, 102)
Crossing obstructions:
01: (1, 0), (2, 2)
01: (1, 0), (3, 2)
012: (0, 1), (0, 1), (2, 2)
012: (0, 1), (0, 1), (3, 2)
021: (0, 1), (0, 1), (4, 1)
021: (2, 2), (2, 2), (3, 2)
021: (2, 2), (3, 2), (3, 2)
102: (2, 2), (2, 2), (3, 2)
102: (2, 2), (3, 2), (3, 2)
Requirement 0:
0: (2, 2)
0: (3, 2)
```

with repr 
```
Tiling(obstructions=(GriddedPerm(Perm((0,)), ((0, 0),)), GriddedPerm(Perm((0,)), ((0, 2),)), GriddedPerm(Perm((0,)), ((1, 1),)), GriddedPerm(Perm((0,)), ((1, 2),)), GriddedPerm(Perm((0,)), ((2, 0),)), GriddedPerm(Perm((0,)), ((2, 1),)), GriddedPerm(Perm((0,)), ((3, 0),)), GriddedPerm(Perm((0,)), ((3, 1),)), GriddedPerm(Perm((0,)), ((4, 0),)), GriddedPerm(Perm((0,)), ((4, 2),)), GriddedPerm(Perm((0, 1)), ((1, 0), (2, 2))), GriddedPerm(Perm((0, 1)), ((1, 0), (3, 2))), GriddedPerm(Perm((1, 0)), ((4, 1), (4, 1))), GriddedPerm(Perm((0, 1, 2)), ((0, 1), (0, 1), (2, 2))), GriddedPerm(Perm((0, 1, 2)), ((0, 1), (0, 1), (3, 2))), GriddedPerm(Perm((0, 2, 1)), ((0, 1), (0, 1), (0, 1))), GriddedPerm(Perm((0, 2, 1)), ((0, 1), (0, 1), (4, 1))), GriddedPerm(Perm((0, 2, 1)), ((1, 0), (1, 0), (1, 0))), GriddedPerm(Perm((0, 2, 1)), ((2, 2), (2, 2), (2, 2))), GriddedPerm(Perm((0, 2, 1)), ((2, 2), (2, 2), (3, 2))), GriddedPerm(Perm((0, 2, 1)), ((2, 2), (3, 2), (3, 2))), GriddedPerm(Perm((0, 2, 1)), ((3, 2), (3, 2), (3, 2))), GriddedPerm(Perm((1, 0, 2)), ((2, 2), (2, 2), (2, 2))), GriddedPerm(Perm((1, 0, 2)), ((2, 2), (2, 2), (3, 2))), GriddedPerm(Perm((1, 0, 2)), ((2, 2), (3, 2), (3, 2))), GriddedPerm(Perm((1, 0, 2)), ((3, 2), (3, 2), (3, 2)))), requirements=((GriddedPerm(Perm((0,)), ((2, 2),)), GriddedPerm(Perm((0,)), ((3, 2),))),), assumptions=())
```

`children`:
```
(Tiling(obstructions=(GriddedPerm(Perm((0,)), ((0, 1),)), GriddedPerm(Perm((0,)), ((1, 0),)), GriddedPerm(Perm((0,)), ((2, 1),)), GriddedPerm(Perm((0, 1)), ((0, 0), (0, 0))), GriddedPerm(Perm((1, 0)), ((2, 0), (2, 0))), GriddedPerm(Perm((0, 2, 1)), ((1, 1), (1, 1), (1, 1))), GriddedPerm(Perm((1, 0, 2)), ((1, 1), (1, 1), (1, 1)))), requirements=((GriddedPerm(Perm((0,)), ((1, 1),)),),), assumptions=(TrackingAssumption((GriddedPerm(Perm((0,)), ((1, 1),)),)),)),)
```

`self.__dict__`:
```
{'col_idx': 2, 'row_idx': None, 'tracked': True, '_ignore_parent': False, '_inferrable': True, '_possibly_empty': False, '_workable': True}
```

**Comment by christianbean** [comment]
I have located the bug. 

The issue was that when finding the fuse parameter, the code did not consider the possibilty of other cells becoming empty. In this tiling we fuse together:
```Requirement 0:
0: (2, 2)
0: (3, 2)```
to become 
```Requirement 0:
0: (2, 2)```
We also fuse the obstructions
```01: (1, 0), (2, 2)
01: (1, 0), (3, 2)```
to become
```
01: (1, 0), (2, 2)
```
Therefore in the child tiling we recognise that cell (1, 0) is empty. This must have been the first instance we saw of this happening.
